docs/arch/repository.rst

```diff
@@ -47,21 +44,21 @@ Chart versioning
 The top level of charts defined in the ``/services`` directory are used only by Argo CD and are never published as Helm charts.
 Their versions are therefore irrelevant.
 The version of each chart is set to ``1.0.0`` because ``version`` is a required field in ``Chart.yaml`` and then never changed.
-Reverting to a previous configuration in this layer of charts is done via a manual revert in Argo CD or by reverting a change in the GitHub repository, not by pointing Argo CD to an older chart.
+It is instead the ``appVersion`` field that is used to point to a particular release of a first-person chart.  Reverting to a previous configuration in this layer of charts is done via a manual revert in Argo CD or by reverting a change in the GitHub repository so that the ``appVersion`` points to an earlier release.  It is **not** done by pointing Argo CD to an older chart.
 
-The second layer of charts that are declared as dependencies are normal, published Helm charts that follow normal Helm semantic versioning conventions.
+Third-party charts are declared as dependencies; they are normal, published Helm charts that follow normal Helm semantic versioning conventions.
 In the case of the ``lsst-sqre/charts`` repository, this is enforced by CI.
 We can then constrain the version of the chart Argo CD will deploy by changing the ``dependencies`` configuration in the top-level chart.
 
 Best practice is for a release of a chart to deploy the latest version of the corresponding service, so that upgrading the chart also implies upgrading the service.
 This allows automatic creation of pull requests to upgrade any services deployed by Argo CD (see `SQR-042 <https://sqr-042.lsst.io/>`__ for more details).
-Charts maintained in lsst-sqre/charts follow this convention (for the most part).
+Charts maintained as first-party charts in Phalanx follow this convention (for the most part).
 Most upstream charts also follow this convention, but some require explicitly changing version numbers in ``values-*.yaml``.
 
 In general, we pin the version of the chart to deploy in the ``dependencies`` metadata of the top-level chart.
 This ensures deterministic cluster configuration and avoids inadvertently upgrading services.
 However, for services still under development, we sometimes use a floating dependency to reduce the number of pull requests required when iterating, and then switch to a pinned version once the service is stable.
 
-There is currently no mechanism to deploy different versions of a chart in different environments.
+There is currently no general mechanism to deploy different versions of a chart in different environments, as appVersion is set in ``Chart.yaml``.
```

Comment:
`appVersion` can be overridden with something like a `image.tag` value in a values file, but that doesn't really get to the point that all environments use the same chart fundamentally.

We do have the ability to test new chart versions by deploying from branches through. Can we link to docs for that? Do those docs exist?

Reply:
Not sure if those docs exist, but if not, I'll write some.  I'll also restructure create-service.

Reply:
I've taken a crack at it.